src/core/core.cpp

```diff
@@ -0,0 +1,62 @@
+#include <vector>
+#include <nlohmann/json.hpp>
+#include <string>
+
+using json = nlohmann::json;
+
+
+struct Product {
+    std::string name;
+    int amount;
+    double price;
+    int whoBoughtID;
+};
+
+struct User {
+    int id;
+    std::string name;
+    std::string phone;
+};
+
+void parseJsonData(const json& jsonData, std::vector<Product>& products, std::vector<User>& users) {
+    for (const auto& receiptData : jsonData["receipt"]) {
+        for (const auto& productData : receiptData["products"]) {
+            Product product;
+            product.name = productData["product"];
+            product.amount = productData["amount"];
+            product.price = productData["price"];
+
+            if (productData.contains("whoBought")) {
+                product.whoBoughtID = productData["whoBought"]["personID"];
+            } else {
+                product.whoBoughtID = productData["whoBought"];
+            }
+
+            products.push_back(product);
+        }
+    }
+
+    for (const auto& userData : jsonData["users"]) {
+        User user;
+        user.id = userData["id"];
+        user.name = userData["name"];
+        user.phone = userData["phone"];
+        users.push_back(user);
+    }
+}
+
+int main() {
+    
+    std::vector<Product> products;
+    std::vector<User> users;
+    std::string jsonData;
+    
+   json parsedData = json::parse(jsonData);
+
+    parseJsonData(parsedData, products, users);
+    }
```

Comment:
does it compile?

Reply:
I can't check now becouse of some problems with my linux system, but it should.
(jsonData) is string with json,The line json parsedData = json::parse(jsonData); is using the nlohmann::json library to parse a JSON string (jsonData) and store the resulting JSON object in a variable named parsedData.